Rules update

abaddon01 opened the pull request:
Indexes added.
Monster, Animal and Vermin individual files added.
* tags for each creature now customised.
Statblock files for monsters, vermin and animals added
Cross linking from creature descriptions to special abilities added.
Properties for all files updated to 1.9 schema.

Commit:
Fix: Complete reconversion of Basic Rules, Divine, Epic, Magic Items, and Psionics from html files through to markdown. All tables should now be converted, headings use correct format, Table of Contents Links now work, tags aligned. Stored under 3.5 Compendium. Next step will be to move legacy md fiels form base directory. Monsters, Spells and Feats are in new format that links spells and feats in monster definitions to the spell/feat description. 3.5 Monster Statblck file can be loaded into fantasy stat blocks to provide 3.5 monsters data using the Basic 3.5 layout included

Commit:
Merge branch 'master' into rules_update

Commit:
fix: feats subheadings set to lvl4 instead of 3

Commit:
missed prereqs :-/

Commit:
fix: error in Skill List internal links- link was escaped

Commit:
fix:removed old markdown files. Updated fantasy stablocks 3.5 monster list and layout. Updated Monsters. Added Special Abilities

Commit:
Added sample json file

Commit:
Updated Special Abilities to include traits (undead traits...) and updated monsters to use them

Commit:
Fixed bug in Skeletons where skills, feats and environment were incorrectly set

Commit:
Monster images and tokens added. Copyright message updated

Commit:
Added indexes including individual monster markdown files - dragons need work

Commit:
Updated statblocks

Commit:
Removed 0hp monsters -True Dragon, Elementals etc. Fixed area in feats missing info in brackets eg weapon focus (bite). added +- to str rep of int values for saves and skills.

Commit:
Updated Special Abilities to reference Poison immunity in the same wasy as the otehrs. Removed status paralysed.

Commit:
Tidy up name-size-type-subtype block

Commit:
Fix bugs in spell generation for monsters. Updated subtypes and file properties and tags

Commit:
Update tags and aliases

Commit:
Added vermion and animals, more tweaks to properties

Commit:
style fixes

Commit:
Added Vulnerabiltiy link to all monsters. Updated Readme with plugin support

Commit:
Updated dot poitn style in Types.

abaddon01 commented:
My thoughts on the two requests: 

I quite like the split out of skills/feats/spells/abiltities in Mr-labb's pull request as well as how they look. These I would happily use in an indexes section.
 - i'd tweak the spells frontmatter a smidge, adding the spell level to the tags section and the same in the level section as below.

```
tags: 
  - Sorcerer 3
  - Wizard 3
Level:
  - Sorcerer 6
  - Wizard 6
  - Water 7
```

I'm not a fan of breaking out the rules themselves into notes and subfolders as Mr-labb has done, I feel it hampers readability and doesn't add much for linking. 
Sections such as basic rules/classes/races/combat I'd prefer kept more in line with the SRD rules as per the reference srd html pages.

I prefer the way I have done the individual monsters/vermins/animals notes to Mr-labb's, I personally don't find the statblock section as useful or as thematic in the rules section. 
I use the statblock version for embedding info for campaign notes, or encounters.

If their concept is to have the statblock section pulled into fantasy statblocks bestiary then either the monsters need to be combined into a json as I have done for loading, or the statblock section needs to be moved into the frontmatter. Which from a viewing the note perspective looks meh.

For my own repo I would use my rules and monsters indexes. and copy Mr-labb's skills/spells/feats/abiltities individual notes into my indexes folder and cross link as appropriate into monsters rules sections and to statblock monsters and players.

I do acknowledge that you have two different pull requests with two different approaches.

mr-labb commented:
I also acknowledge that we have two different pull requests with two different approaches, and whit your coment I thnik we may achive an agement for the main vault


> Sections such as basic rules/classes/races/combat I'd prefer kept more in line with the SRD rules as per the reference srd html pages.

No probleme for me, for my own repo I would keep it separted


> i'd tweak the spells frontmatter a smidge, adding the spell level to the tags section and the same in the level section as below

I'am not sure what is the advantage of having them in both tags and level
No matter whether we use Tags or Level, we can use them in Base to filter. However, I find it more useful to have a general tag — for example, Wizard — to eventually filter everything related to a class, rather than putting everything into the tags.

> I prefer the way I have done the individual monsters/vermins/animals notes

No problem on my end — it’s just a different way of organizing things.
That said, I don’t really see the benefit, since Base already allows filtering specifically for vermins or animals.

> don't find the statblock section as useful or as thematic

When converting monsters, I included the Statblock simply because that was the plan as described in the README. For now, I’m not using the Initiative Tracker or the Dice Roller, so I’m not sure how useful they are, although I do think they look good and easy to read.

Your comnet give me an idea and I have added headings to each entry to frame the Statblock, ## Statblock just before and ## Description right after. That way, a call like ![[Monster#Statblock]] could be used to embed info for campaign notes or encounters.


That said, in my personal Vault, I have a bit over 2,000 additional entries from other 3.5 manuals that don’t include statblocks. For now, I don’t see the need to convert them.

One thing to note for monsters — regardless of the format chosen — is that it would be useful to insert a new property Call Thumbnail or image to enable the card view feature in Base. It looks very good.

I’ve already added this in the upload I did today.

The release of version 1.9 on Monday also helped me identify a number of errors in the properties and statblocks of monsters, so this new version is more complete and reliable.

abaddon01 commented:
I also acknowledge that we have two different pull requests with two different approaches, and whit your coment I thnik we may achive an agement for the main vault


> I also acknowledge that we have two different pull requests with two different approaches, and whit your coment I thnik we may achive an agement for the main vault
> 
> > Sections such as basic rules/classes/races/combat I'd prefer kept more in line with the SRD rules as per the reference srd html pages.
> 
> No probleme for me, for my own repo I would keep it separted
yes i think they should be separate

> 
> > i'd tweak the spells frontmatter a smidge, adding the spell level to the tags section and the same in the level section as below
> 
> I'am not sure what is the advantage of having them in both tags and level No matter whether we use Tags or Level, we can use them in Base to filter. However, I find it more useful to have a general tag — for example, Wizard — to eventually filter everything related to a class, rather than putting everything into the tags.
> 
Good point, I agree with your reasoning - general tag, make the level specific.


> > I prefer the way I have done the individual monsters/vermins/animals notes
> 
> No problem on my end — it’s just a different way of organizing things. That said, I don’t really see the benefit, since Base already allows filtering specifically for vermins or animals.

I was thinking more that I don't like the blending of the statblock at the start and then the rest of the information in a different style as per "don't find the statblock section as useful or as thematic"

> > don't find the statblock section as useful or as thematic
> 
> When converting monsters, I included the Statblock simply because that was the plan as described in the README. For now, I’m not using the Initiative Tracker or the Dice Roller, so I’m not sure how useful they are, although I do think they look good and easy to read.
> 
Dice roller I haven't found particularly useful, but the initiative tracker and being able to embed encounters inside campaign notes has been useful. I've been using the encounter functionality in games and it has been easier then doing it by paper with my inevitable - sorry X I missed you :-/ 

> Your comnet give me an idea and I have added headings to each entry to frame the Statblock, ## Statblock just before and ## Description right after. That way, a call like ![[Monster#Statblock]] could be used to embed info for campaign notes or encounters.
> 
I'd have it the other way around - description then statblock. As an alternative idea I have a seperate pc_name.stat note that I use for PC's and NPC's stat blocks. I then call that in the players/npc's description note. ![[pc_name.stat]], migth work nicely for the monster stats?

> That said, in my personal Vault, I have a bit over 2,000 additional entries from other 3.5 manuals that don’t include statblocks. For now, I don’t see the need to convert them.
>
impressive!
 
> One thing to note for monsters — regardless of the format chosen — is that it would be useful to insert a new property Call Thumbnail or image to enable the card view feature in Base. It looks very good.
> 

Sounds like an excellent idea.

> I’ve already added this in the upload I did today.
> 
> The release of version 1.9 on Monday also helped me identify a number of errors in the properties and statblocks of monsters, so this new version is more complete and reliable.

I haven't had a play with 1.9 as yet.

My thinking was taking your abiltities/skills/spells folders and moving them to an indexes folder, and deleting my Spell List/trait List... notes, as your versions supercede them.
 I suppose the same would work with what you have done in the rules sections - which is making them more folders of the individual components (classes races etc) rather than a book format - having them in an indexes section?

I think it would be worth having a branch in DnD-3.5-SRD-Markdown to work in rather than master to try out some different combinations?

mr-labb commented:
>I'd have it the other way around – description then statblock

In principle, I mostly agree. My concern is that with open source, the descriptions are quite limited, and what we mostly see are the creatures' special qualities rather than a true narrative description.
A simple Python script could easily move the statblock to the end of the file, since it's clearly delimited by '''Statblock and '''.

>As an alternative idea I have a separate pc_name.stat

Regarding the pc_name.stat option, I understand what you're trying to achieve, but I'm still skeptical about its usefulness in the context of the SRD. Just like placing the statblock at the end, it tends to leave very little room for descriptive content, and in terms of rules, the value often lies in the stats themselves.
One possibility might be to move the statblock after the description, just before the COMBAT section. Unfortunately, not all monsters follow the same structure—though most do. Again, a Python script could probably handle it, but it would require reviewing each entry individually to avoid errors.
Maybe I'm misunderstanding your reasoning. If it's more about style, are you suggesting formatting the information in a table that resembles what's found in the official books? If so, converting to Markdown tables is definitely doable, but it's a significant amount of work I’m not ready to commit to—though I don’t see any major downside to it.
For integration with an initiative tracker, I agree that the pc_name.stat file would be necessary. But I find it a bit redundant and worry it could lead to discrepancies between data sources. If that’s the intended effect, maybe it would be more useful to explore whether Fantasy Statblock’s style can be adjusted to better match 3rd edition formatting.
As for the "indexes folder" you mentioned, I’m not quite sure what you mean. I’m just getting started with Obsidian and have only used the basic features so far.
Same goes for GitHub—I’m still learning. I’m not sure whether we need to merge one of our forks or if it’s possible to create a second copy and work from there.
My fork is completely separate from my personal vault. The SRD is fairly limited, and I didn’t want to risk including any content not covered by open source, so it should be safe to work from.

abaddon01 commented:
I've been fiddling with ideas in the background, I'll create a branch in my repo that incorporates your files to give you a feel for what I'm thinking so that you can critque it. 
It will look something like this

** Top level **
- 3.5 Compendium (based off the HTML SRD layout closer to a book format - better for reading)
   -  Basic Rules (my files)
   - Divine (my files)
   - Epic (my files)
   - Magic Items ( your files, having the cross link to spells is really useful)
   - Monsters (my files markdown only)
   - Psionics (my files )
   - Spells (my files)
-  3.5 Lists (mostly your files where you have separated each feat/Traits/Spell as a separate file - better for linking)
   - Classes (your files)
   - Feats (your files)
   - Races (your files)
   - Skills (Your files )
   - Spells (Your files)
   - Special Abiltiies (your files)
   - Traits and Types (your files)
   - Besitiary
      - tokens and images (sourced from dndbeyond and other obsidian repos) 
      -  vermin ( my files )
      -  animals ( my files)
      - monsters (my files)
 - 3.5 Statblocks  (json files for loading into fantasy statblocks )
    - 3.5 Animal Statblocks.json
    - 3.5 Vermin Statblocks.json
    - 3.5 Monster Statblocks.json
    - Basic 3.5e Layout.json
 
I'm still fiddling with the monster file formating to go into the 3.5 Lists Folder
I'm leaning towards just having the statblocks and no other stuff - with the statblocks including the description inside the statblock itself so that they are a complete representation of the creature.
 
![aboleth|400](https://github.com/user-attachments/assets/9d8a44e2-a9a4-4b83-99c4-d0cd2937afff)

agree with the image property - once i know how to make it work :-)

abaddon01 commented:
Will work on vault in slow time.
